streamer/input_configuration.py

```diff
@@ -315,10 +315,43 @@ def get_input_args(self) -> List[str]:
   def get_resolution(self) -> bitrate_configuration.VideoResolution:
     return bitrate_configuration.VideoResolution.get_value(self.resolution)
 
+class SinglePeriod(configuration.Base):
+  """An object repersenting one optional video stream and multiple audio and text streams"""
+
+  inputs = configuration.Field(List[Input], required=True).cast()
 
 class InputConfig(configuration.Base):
   """An object representing the entire input config to Shaka Streamer."""
 
-  inputs = configuration.Field(List[Input], required=True).cast()
-  """A list of Input objects, one per input stream."""
+  multiperiod_inputs_list = configuration.Field(List[SinglePeriod]).cast()
+  """A list of SinglePeriod objects"""
+
+  inputs = configuration.Field(List[Input]).cast()
+  """A list of Input objects"""
+
+  def __init__(self, dictionary: Dict[str, Any]):
+    """A constructor to check that either inputs or mutliperiod_inputs_list is provided,
+    and produce a helpful error message in case both or none are provided.
+    
+    We need these checks before passing the input dictionary to the configuration.Base constructor,
+    because it does not check for this 'exclusive or-ing' relationship between fields.
+    """
+
+    assert isinstance(dictionary, dict), """Malformed Input Config File,
+    See some examples at https://github.com/google/shaka-streamer/tree/master/config_files.
+    """
+
+    if dictionary.get('inputs') and dictionary.get('multiperiod_inputs_list'):
+      raise configuration.ConflictingFields(
+        InputConfig, 'inputs', configuration.Field(List[Input]), ['inputs', 'multiperiod_inputs_list']
+      )
+    
+    # Because these fields are not marked as required at the class level
+    # , we need to check ourselves that one of them is provided.
+    if dictionary.get('inputs') is None and dictionary.get('multiperiod_inputs_list') is None:
+      raise configuration.MissingRequiredField(
+        InputConfig, 'either inputs or multiperiod_inputs_list', configuration.Field(Dict[str, Any])
```

Comment:
I'm not certain about these errors where you make up configuration.Field values with bogus types like Any.  I wonder if this is why you had to add "Any" handling to the type formatter.

How about this:

1. `ConflictingFields` takes the two field names and the class and uses the names to extract the declared types
2. Create a `MissingRequiredExclusiveFields` that does the same, but with a message that says "exactly one of" ... "is required"
3. Use both field names and types in the error message strings of those error types